[Mobile]: Improving accessibility on Image Block (deselected)

etoledom opened the pull request:
This PR improves accessibility on the Image Block.

To test:
Please refer to the [gutenberg-mobile side PR](https://github.com/wordpress-mobile/gutenberg-mobile/pull/794).

etoledom committed:
Adding accessibility properties to MediaPlaceholder and ImageEdit

etoledom committed:
Fix lint issues

Tug commented on packages/block-library/src/image/edit.native.js:
```diff
@@ -327,7 +327,14 @@ class ImageEdit extends React.Component {
+				accessible={ ! isSelected }
+				accessibilityLabel={ __( 'Image block' ) + '.' + alt }
+				accessibilityRole={ 'button' }
+				accessibilityHint={ __( 'Doulbe tap to edit the image' ) }
```
`Doulbe` => `Double`

Tug commented on packages/block-editor/src/components/media-placeholder/index.native.js:
```diff
@@ -16,7 +16,12 @@ import styles from './styles.scss';
 	return (
-		<TouchableWithoutFeedback onPress={ props.onMediaOptionsPressed }>
+		<TouchableWithoutFeedback
+			accessibilityLabel={ __( 'Image block. Empty ' ) }
```
I guess we can reuse the 'Image block' string here. `Image block. Empty` as a whole does not make much sense for translators:
```
accessibilityLabel={ __( 'Image block' ) + '. ' + __( 'Empty' ) }
```

Tug commented on packages/block-library/src/image/edit.native.js:
```diff
@@ -327,7 +327,14 @@ class ImageEdit extends React.Component {
-			<TouchableWithoutFeedback onPress={ this.onImagePressed } disabled={ ! isSelected }>
+			<TouchableWithoutFeedback
+				accessible={ ! isSelected }
+				accessibilityLabel={ __( 'Image block' ) + '.' + alt }
```
It could use a space after the period:
```
__( 'Image block' ) + '. ' + alt
```

etoledom committed:
Fix typo

etoledom committed:
Merge branch 'rnmobile/accessibility-mark-ii' of https://github.com/WordPress/gutenberg into rnmobile/accessibility-mark-ii

etoledom committed:
Better accessibility labels

etoledom commented:
Thank you for the review @Tug ! 
Added the requested changes. Nice catches :)

SergioEstevao committed:
Disable lists. (#14844)

etoledom committed:
Merge commit '7b89592' into rnmobile/accessibility-mark-ii

etoledom committed:
Fix scroll with VoiceOver ON.

For some reason, VoiceOver wouldn't scroll down the block list.
The breaking change was located at the removal of this css property `font-family: $default-regular-font;` on the nextpage block.

Addinb back that property to be able to scroll the blocks list again using VoiceOver.

etoledom committed:
Setting accessibilityStates properly to avoid Android crash

etoledom committed:
Merge branch 'master' into rnmobile/accessibility-mark-ii

etoledom committed:
Remove vscode settings file (added by mistake)

etoledom commented on packages/block-library/src/nextpage/edit.native.js:
```diff
@@ -14,20 +14,21 @@ import { __ } from '@wordpress/i18n';
-	// that React Native wasn't able to handle (adding a fallback generic font family).
-	const textStyle = {
-		...styles[ 'block-library-nextpage__text' ],
-		fontFamily: 'System',
```
This was creating an issue that disabled scrolling with VoiceOver active.
I'm not sure the specific reason, but rolling back this change fixed the issue.

etoledom commented:
I cherry-picked a commit to fix an issue scrolling with VoiceOver active: https://github.com/WordPress/gutenberg/pull/14713/commits/14652d67401060aeb3c20bdb4d4f0c0aa77df7dc.

That commit also brought ScreenReader improvements for the NextPage block.

@Tug this is ready for another check. Adding @pinarol as reviewer too.

etoledom committed:
Fix lint issues

pinarol commented on packages/block-library/src/image/edit.native.js:
```diff
@@ -328,7 +328,14 @@ class ImageEdit extends React.Component {
-			<TouchableWithoutFeedback onPress={ this.onImagePressed } disabled={ ! isSelected }>
+			<TouchableWithoutFeedback
+				accessible={ ! isSelected }
+				accessibilityLabel={ __( 'Image block' ) + '. ' + alt }
```
the dot here can be localized also

pinarol commented on packages/block-editor/src/components/media-placeholder/index.native.js:
```diff
@@ -16,7 +16,12 @@ import styles from './styles.scss';
 	return (
-		<TouchableWithoutFeedback onPress={ props.onMediaOptionsPressed }>
+		<TouchableWithoutFeedback
+			accessibilityLabel={ __( 'Image block' ) + '. ' + __( 'Empty' ) }
```
the dot here can be localized also

pinarol commented on packages/block-library/src/nextpage/edit.native.js:
```diff
@@ -14,19 +14,20 @@ import { __ } from '@wordpress/i18n';
-		<View style={ styles[ 'block-library-nextpage__container' ] }>
+		<View
+			accessible
+			accessibilityLabel={ __( 'Page break block' ) + '. ' + accessibilityTitle }
```
the dot here can be localized also

pinarol commented:
I've tried adding accesibility label for Image itself also, I think it felt better to be able to hear the image info independent from its selected state. 

```
                                    <ImageBackground
                                        accessible={ true }
                                        accessibilityLabel={ __( ‘Image’ ) + ‘. ’ + alt }
```

pinarol reviewed (approved):
Thanks! @etoledom 🎉 
I think it'd be ok for me to approve since localization will be very simple and the extra accessibility label will be a part of selected version's PR 

:shipit:

etoledom committed:
Update accessibility labels to use sprintf

etoledom committed:
Fix lint issues

etoledom commented:
Thank you!
